modules/internal/ChapelIO.chpl

```diff
@@ -465,7 +465,7 @@ module ChapelIO {
         if st == QIO_AGGREGATE_FORMAT_JSON {
           start = new ioLiteral("{");
         } else if st == QIO_AGGREGATE_FORMAT_CHPL {
-          start = new ioLiteral("new " + typeToString(t) + "(");
+          start = new ioLiteral("new " + typeToString(t).c_str() + "(");
```

Comment:
Actually, this is a bit wonky right now.

`: c_string` does work on param, but will fail for a normal value (and this I think this needs to happen for functionResolution to be somewhat sane).

`.c_str()` will work on both, but it does not have a param version. So this `new ioLiteral(...)` call is actually going to do runtime string concatenation right now.


I've added a param version of `.c_str()` that should make things a bit better. Users should be using `.c_str()` to get a `c_string` from a `string` in all cases.